### PR TITLE
947 example for ide secir model not working as expected

### LINKED ISSUE #947: Example for IDE-SECIR model not working as expected
### Motivation / Current Behaviour

Currently, if the step size dt in the simple model is changed to e.g. 0.001, the results include negative numbers.

### Enhancement description

Rework the example such that the results are reasonable for smaller step sizes.

### Additional context

_No response_

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

### PR DESCRIPTION
**Pull request by lenaploetzke**
# Changes and Information

Please **briefly list the changes** (main added features, changed items, or corrected bugs) made:

- Changed the initial transitions so that when using a smaller step size, the resulting number of people in a compartment is no longer negative.
- small corrections of the comments

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful (with and without OpenMP)
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [ ] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development) and ran and posted a local test above from before and after development to ensure performance is monitored.

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes, code coverage and performance is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] On merge, add 2-5 lines with the changes (main added features, changed items, or corrected bugs) to the merge-commit-message. This can be taken from the **briefly-list-the-changes** above (best case) or the separate commit messages (worst case).

Closes #947.

### CODE REVIEW

File: cpp/examples/ide_secir.cpp [hunk 1]
```diff
@@ -26,7 +26,6 @@
 #include "memilio/math/eigen.h"
 #include "memilio/utils/time_series.h"
 #include "memilio/epidemiology/uncertain_matrix.h"
```
Comment (HenrZu) [inline cpp/examples/ide_secir.cpp:28]:
```suggestion
```

This include is superfluous. Check for others

File: cpp/examples/ide_secir.cpp [hunk 2]
```diff
@@ -98,6 +98,6 @@ int main()
     sim.advance(tmax);
 
     sim.get_result().print_table({"S", "E", "C", "I", "H", "U", "R", "D "}, 16, 8);
```
Comment (HenrZu) [inline cpp/examples/ide_secir.cpp:100]:
```suggestion
       auto interpolated_results = mio::interpolate_simulation_result(sim.get_result());
       interpolated_results.print_table({"S", "E", "C", "I", "H", "U", "R", "D "}, 16, 8);
```
dt is 0.01 so you are printing a lot of (maybe uninteresting) results.

File: cpp/examples/ide_secir.cpp [hunk 3]
```diff
@@ -35,14 +34,14 @@ int main()
     ScalarType tmax   = 10;
     ScalarType N      = 10000;
     ScalarType deaths = 13.10462213;
-    ScalarType dt     = 1;
+    ScalarType dt     = 0.01;
```
Comment (HenrZu) [inline cpp/examples/ide_secir.cpp:37]: maybe write a short comment, what you have learned form other choices of dt.

**Review by HenrZu — CHANGES_REQUESTED** [review]

File: cpp/examples/ide_secir.cpp (hunk 1 above)
Comment (lenaploetzke) [reply cpp/examples/ide_secir.cpp:28]: I am using an UncertainContactMatrix in line 86. Why is this superflous?

File: cpp/examples/ide_secir.cpp (hunk 2 above)
Comment (lenaploetzke) [reply cpp/examples/ide_secir.cpp:100]: This is actually a nice tool!

File: cpp/examples/ide_secir.cpp (hunk 3 above)
Comment (lenaploetzke) [reply cpp/examples/ide_secir.cpp:37]: The study of the order of convergence is not yet finished, so I don't think it's appropriate here.
The point was simply to allow different step sizes in this example. Or did you think of something else?

**Review by HenrZu — APPROVED** [review]

**Pull request merged**